src/internal/puck/components/ThemeHeader.tsx

```diff
@@ -64,29 +64,43 @@ export const ThemeHeader = (props: ThemeHeaderProps) => {
       <div className="header-center"></div>
       <div className="actions">
         <ClearLocalChangesButton
-          disabled={themeHistory?.history.length === 1}
+          disabled={themeHistory?.histories?.length === 1}
           onClearLocalChanges={() => {
             clearThemeHistory();
             if (themeConfig) {
-              updateThemeInEditor(themeHistory?.history[0], themeConfig);
+              updateThemeInEditor(
+                themeHistory?.histories?.[0]?.state
+                  ?.data as unknown as SavedTheme,
+                themeConfig
+              );
             }
             setThemeHistory({
-              history: [themeHistory?.history[0]],
-              index: 0,
+              histories: [
+                {
+                  id: "root",
+                  state: { data: themeHistory?.histories?.[0]?.state?.data },
+                },
+              ],
             });
           }}
         />
         {!isDevMode && (
           <Button
             variant="secondary"
-            disabled={themeHistory?.history.length === 1}
+            disabled={themeHistory?.histories?.length === 1}
             onClick={async () => {
               await onPublishTheme();
               setThemeHistory({
-                history: [
-                  themeHistory?.history[themeHistory?.history.length - 1],
+                histories: [
+                  {
+                    id: "root",
```

Comment:
I don't think we want to rest the id. This only made sense for Puck. We already have a generated id right? All we're doing here is publishing and then setting the themeHistories back to the last item, which should have a hash already from when it was generated.